ui/StatusQ/src/StatusQ/Components/StatusDateInput.qml

```diff
@@ -0,0 +1,263 @@
+            monthInput.placeholderText: qsTr("mm")
+            yearInput.placeholderText: qsTr("yyyy")
+            presetTimestamp: fromTimestamp
+        }
+   \endqml
+
+   For a list of components available see StatusQ.
+*/
+
+Control {
+    id: root
+
+    /*!
+        \qmlproperty alias StatusDateInput::dateInput
+        This property exposes the dateInput for customisation
+    */
+    property alias dateInput: editDate
+    /*!
+        \qmlproperty alias StatusDateInput::monthInput
+        This property exposes the monthInput for customisation
+    */
+    property alias monthInput: editMonth
+    /*!
+        \qmlproperty alias StatusDateInput::yearInput
+        This property exposes the yearInput for customisation
+    */
+    property alias yearInput: editYear
+
+    /*!
+        \qmlproperty string StatusDateInput::nowText
+        This property holds the now text to be shown on the eidget
+    */
+    property string nowText
+    /*!
+        \qmlproperty double StatusDateInput::presetTimestamp
+        This property holds the timestamp shosen before entering the popup
+    */
+    property double presetTimestamp: Date.now()
+    /*!
+        \qmlproperty bool StatusDateInput::isEditMode
+        This property can turn and off the edit mode for the input
+    */
+    property bool isEditMode: false
+    /*!
+        \qmlproperty bool StatusDateInput::showBackground
+        This property helps turning of the background of the input
+    */
+    property bool showBackground: true
+    /*!
+        \qmlproperty var StatusDateInput::newDate
+        Represents the newly set date in the input
+    */
+    property var newDate
+    /*!
+        \qmlproperty string StatusDateInput::errorMessage
+        This property to assign errorMessage for the date input
+    */
+    property string errorMessage
+    /*!
+        \qmlproperty bool StatusDateInput::isEditMode
+        This property exposes if the input has valid date
+    */
+    readonly property bool valid: editDate.acceptableInput && editMonth.acceptableInput && editYear.acceptableInput
+    /*!
+        \qmlproperty bool StatusDateInput::hasChange
+        This property exposes if the input has been modified by user
+    */
+    readonly property bool hasChange: d.presetDate.valueOf() !== newDate.valueOf()
+    /*!
+        \qmlproperty bool StatusDateInput::hasChange
+        This property exposes if the input has been modified by user
+    */
+    property int supportedStartYear: 0
+
+    /*!
+        \qmlmethod
+        This function resets the input's text
+    */
+    function reset() {
+        d.presetDate = d.getDateWithoutTime(presetTimestamp)
+    }
+    /*!
+        \qmlmethod
+        This function sets the active focus to edit date
+    */
+    function forceActiveFocus() {
+        editDate.forceActiveFocus()
+        editDate.cursorPosition = 0
+    }
+
+    QtObject {
+        id: d
+        readonly property string separator: "/"
```

Comment:
Nope... the separator is locale specific; but there's a gotcha, `QLocale` doesn't expose it. Also the order of the fields is locale sensitive

Reply:
I guess cause this is a date picker 
![image](https://github.com/status-im/status-desktop/assets/60327365/e714bf30-4bc0-4e37-9c94-586172811209)
i think it would be okay if the separator is "/" always in this case, its not a text field we are trying to show

Reply:
> Yeah I have to agree but as this is very hard to get right, what about we use the `StatusDatePicker` instead for the selection? It would look the same but offer a calendar popup to choose the date (we use this in other places already). Wdyt @benjthayer? This will allow us to avoid any confusion about the order of elements here and also format the date correctly. Looks like this:
> 
> ![image](https://user-images.githubusercontent.com/5377645/241924496-4332a3e9-3b8b-4fa9-8067-ade2808bed89.png)

Thanks @caybro. To think that through a little more, is your idea that we modify the interaction so that the user can only edit the date via the (existing) date picker flyout and we block them from editing the date string directly? We could do this and it would avoid a number of validation cases eg where the user can enter letters or emojis into the field! We would just need to include a calendar icon and treat it like a dropdown as opposed to an input to avoid confusion.

My only concern is that it is less efficient than the user directly entering the date string.

The alternative is to have the prompt inline with the field label eg: 'From:    DD/MM/YYYY' with the format on the right in a lighter grey. That would clarify the ordering based on the user's locale and would just leave the issue that the inclusion of the `/` may not be appropriate for those whose locale dictates this should be a `.` instead.

WDYT?

https://www.figma.com/file/FkFClTCYKf83RJWoifWgoX/Wallet-v2?type=design&node-id=12651-323708&t=uGwQJyFaAx4MXuaO-4

Reply:
It's being tracked here: https://github.com/status-im/status-desktop/issues/10862
I'd say @Khushboo-dev-cpp you can either use the current StatusDatePicker as it is now (and I'll adjust the code if needed later), or just keep this PR intact and we can change the impl when I'm done with the date picker changes, your choice :)